### PR TITLE
External and internal class renaming

### PR DESCRIPTION
**Pull request by liran-funaro**
- Remove redundant "Oak" prefix to all internal classes
 - Changed attached/detached to scoped/unscoped
 - Add OakBuffer interface as a parent of OakScopedReadBuffer and OakUnscopedBuffer

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: core/src/main/java/com/oath/oak/BlockMemoryAllocator.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
  * any thread.
  * Note that two allocations cannot be merged into a single allocation, and an allocation currently is not split.
  */
-interface OakBlockMemoryAllocator {
+interface BlockMemoryAllocator {
```
Comment (sanastas) [inline core/src/main/java/com/oath/oak/BlockMemoryAllocator.java:16]: This was initially an outside interface... Before the package was removed, so it is a candidate to be exposed outside in the future...

File: core/src/main/java/com/oath/oak/InternalOakMap.java [hunk 2]
```diff
@@ -1415,50 +1415,50 @@ public T next() {
...
 
-    class EntryStreamIterator extends Iter<Map.Entry<OakDetachedBuffer, OakDetachedBuffer>>
-            implements Map.Entry<OakDetachedBuffer, OakDetachedBuffer> {
+    class EntryStreamIterator extends Iter<Map.Entry<OakUnscopedBuffer, OakUnscopedBuffer>>
+            implements Map.Entry<OakUnscopedBuffer, OakUnscopedBuffer> {
 
-        private final OakDetachedReadBuffer<KeyBuffer> key = new OakDetachedReadBuffer<>(new KeyBuffer());
-        private final OakDetachedReadBuffer<ValueBuffer> value =
-                new OakDetachedReadBuffer<>(new ValueBuffer(valueOperator));
+        private final UnscopedBuffer<KeyBuffer> key = new UnscopedBuffer<>(new KeyBuffer());
```
Comment (sanastas) [inline core/src/main/java/com/oath/oak/InternalOakMap.java:1427]: So there is no unscoped-read/write- buffer anymore?

File: core/src/main/java/com/oath/oak/OakBuffer.java [hunk 3]
```diff
@@ -12,17 +12,8 @@
...
- * The behaviour of this buffer outside their attached scope is undefined.
- * Such callback method might be the application's serializer and comparator, or a lambda function that can
- * read/store/update the data.
- * This access reduces unnecessary copies and deserialization of the underlying data.
- * In their intended context, the user does not need to worry about concurrent accesses and memory management.
- * Using these buffers outside their intended context may yield unpredicted results, e.g., reading non-consistent data
- * and/or irrelevant data.
  */
-public interface OakReadBuffer {
+public interface OakBuffer {
```
Comment (sanastas) [inline core/src/main/java/com/oath/oak/OakBuffer.java:16]: Didn't we say it is not going to be public?

Comment (liran-funaro) [reply core/src/main/java/com/oath/oak/OakBuffer.java:16]: Correct. Will fix.

File: core/src/main/java/com/oath/oak/InternalOakMap.java (hunk 2 above)
Comment (liran-funaro) [reply core/src/main/java/com/oath/oak/InternalOakMap.java:1427]: Unscoped write buffer never existed before.
Before, I added "read" to the unscoped buffer name. But I think it is redundant.
Don't you think?

File: core/src/main/java/com/oath/oak/OakBuffer.java (hunk 3 above)
Comment (liran-funaro) [reply core/src/main/java/com/oath/oak/OakBuffer.java:16]: After working on Druid, I realized I need a common interface between all types of buffers.
So I think this should remain public.

Comment (liran-funaro) [reply core/src/main/java/com/oath/oak/OakBuffer.java:16]: This is a way for the developer to state: "I don't care what kind of buffer I get (scoped/unscoped/whatever)".

File: core/src/main/java/com/oath/oak/InternalOakMap.java (hunk 2 above)
Comment (sanastas) [reply core/src/main/java/com/oath/oak/InternalOakMap.java:1427]: Unscoped buffer indeed exists for read only. As it is internally used name I am OK with leaving it like this. Maximum we will change it later.

File: core/src/main/java/com/oath/oak/OakBuffer.java (hunk 3 above)
Comment (sanastas) [reply core/src/main/java/com/oath/oak/OakBuffer.java:16]: Hmm... OK...

Comment (sanastas) [reply core/src/main/java/com/oath/oak/OakBuffer.java:16]: Then something need to be said about it in README PR#88

**Pull request merged**